ui/app/adapters/database/credential.js

```diff
@@ -0,0 +1,16 @@
+import ApplicationAdapter from '../application';
+
+export default ApplicationAdapter.extend({
+  namespace: 'v1',
+
+  fetchByQuery(store, query) {
+    const { backend, roleType, secret } = query;
+    let creds = roleType === 'static' ? 'static-creds' : 'creds';
+    return this.ajax(`${this.buildURL()}/${backend}/${creds}/${secret}`, 'GET').then(resp => {
+      return resp;
+    });
+  },
```

Comment:
Can any variables you are passing into the TemplateLiteral in these ajax calls contain URL unsafe characters? I'm guessing possibly no, but mainly unsure about what `secret` could be/contain?

Reply:
Oh also, tiny detail, you don't need the extra `then` here if you are just returning the `resp`

Reply:
secret is the user-provided name of the role, so that should for sure be encoded. Backend too, thanks for the catch!